### ISSUE TITLE
quarkus-smallrye reactive Kafka + strimzi AVRO fails on OCP4.6/9

### ISSUE THREAD

**Issue by pjgg** [quarkusio/quarkus#21534]
### Describe the bug

Quarkus Version: `999-SNAPSHOT` 
Openshift: `4.6/4.9`
Extensions that are involved: 
```
- quarkus-smallrye-reactive-messaging-kafka
- quarkus-apicurio-registry-avro
- quarkus-avro
```
**Works as expected with Quarkus 2.3.0.Final**

When I produce some events serialized with AVRO in a Quarkus app deployed into Openshift, we got the following error:

 ```
`8:28:39,116 SRMSG18212: Message org.eclipse.microprofile.reactive.messaging.Message$8@5b759197 was not sent to Kafka topic 'stock-price' - nacking message: java.lang.IllegalStateException: Retries exhausted : 10 attempts against 1637173737866/1637173737232 expiration
	at io.smallrye.mutiny.helpers.ExponentialBackoff.lambda$randomExponentialBackoffFunctionExpireAt$2(ExponentialBackoff.java:113)
	at io.smallrye.context.impl.wrappers.SlowContextualFunction.apply(SlowContextualFunction.java:21)
	at io.smallrye.mutiny.groups.MultiOnItem.lambda$transformToUni$6(MultiOnItem.java:256)
	at io.smallrye.mutiny.operators.multi.MultiFlatMapOp$FlatMapMainSubscriber.onItem(MultiFlatMapOp.java:176)
	at io.smallrye.mutiny.subscription.MultiSubscriber.onNext(MultiSubscriber.java:61)
	at io.smallrye.mutiny.operators.multi.MultiFlatMapOp$FlatMapMainSubscriber.tryEmit(MultiFlatMapOp.java:228)
	at io.smallrye.mutiny.operators.multi.MultiFlatMapOp$FlatMapInner.onItem(MultiFlatMapOp.java:597)
	at io.smallrye.mutiny.subscription.MultiSubscriber.onNext(MultiSubscriber.java:61)
	at io.smallrye.mutiny.converters.uni.UniToMultiPublisher$UniToMultiSubscription.onItem(UniToMultiPublisher.java:82)
	at io.smallrye.mutiny.operators.uni.builders.DefaultUniEmitter.complete(DefaultUniEmitter.java:36)
	at io.smallrye.mutiny.groups.UniRetry.lambda$addPredicateToBackoffFactory$0(UniRetry.java:108)
	at io.smallrye.context.impl.wrappers.SlowContextualConsumer.accept(SlowContextualConsumer.java:21)
	at io.smallrye.mutiny.operators.uni.builders.UniCreateWithEmitter.subscribe(UniCreateWithEmitter.java:22)
	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
	at io.smallrye.mutiny.converters.uni.UniToMultiPublisher$UniToMultiSubscription.request(UniToMultiPublisher.java:63)
	at io.smallrye.mutiny.operators.multi.MultiFlatMapOp$FlatMapInner.onSubscribe(MultiFlatMapOp.java:591)
	at io.smallrye.mutiny.converters.uni.UniToMultiPublisher.subscribe(UniToMultiPublisher.java:24)
	at io.smallrye.mutiny.groups.MultiCreate$1.subscribe(MultiCreate.java:156)
	at io.smallrye.mutiny.operators.multi.MultiFlatMapOp$FlatMapMainSubscriber.onItem(MultiFlatMapOp.java:191)
	at io.smallrye.mutiny.subscription.MultiSubscriber.onNext(MultiSubscriber.java:61)
	at io.smallrye.mutiny.operators.multi.processors.UnicastProcessor.drainWithDownstream(UnicastProcessor.java:103)
	at io.smallrye.mutiny.operators.multi.processors.UnicastProcessor.drain(UnicastProcessor.java:134)
	at io.smallrye.mutiny.operators.multi.processors.UnicastProcessor.onNext(UnicastProcessor.java:200)
	at io.smallrye.mutiny.operators.multi.processors.SerializedProcessor.onNext(SerializedProcessor.java:105)
	at io.smallrye.mutiny.subscription.SerializedSubscriber.onItem(SerializedSubscriber.java:74)
	at io.smallrye.mutiny.subscription.MultiSubscriber.onNext(MultiSubscriber.java:61)
	at io.smallrye.mutiny.operators.multi.MultiRetryWhenOp$RetryWhenOperator.onFailure(MultiRetryWhenOp.java:116)
	at io.smallrye.mutiny.subscription.MultiSubscriber.onError(MultiSubscriber.java:73)
	at io.smallrye.mutiny.converters.uni.UniToMultiPublisher$UniToMultiSubscription.onFailure(UniToMultiPublisher.java:92)
	at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onFailure(UniOperatorProcessor.java:48)
	at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onFailure(UniOperatorProcessor.java:48)
	at io.smallrye.mutiny.operators.uni.builders.DefaultUniEmitter.fail(DefaultUniEmitter.java:52)
	at io.smallrye.mutiny.operators.uni.builders.UniCreateWithEmitter.subscribe(UniCreateWithEmitter.java:26)
	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
	at io.smallrye.mutiny.operators.uni.UniRunSubscribeOn.lambda$subscribe$0(UniRunSubscribeOn.java:27)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.IllegalArgumentException: port p must be in range 0 <= p <= 65535
	at io.vertx.core.impl.Arguments.requireInRange(Arguments.java:44)
	at io.vertx.core.net.SocketAddress.inetSocketAddress(SocketAddress.java:46)
	at io.vertx.core.http.impl.HttpClientImpl.doRequest(HttpClientImpl.java:533)
	at io.vertx.core.http.impl.HttpClientImpl.request(HttpClientImpl.java:418)
	at io.vertx.ext.web.client.impl.HttpContext.handleCreateRequest(HttpContext.java:530)
	at io.vertx.ext.web.client.impl.HttpContext.execute(HttpContext.java:375)
	at io.vertx.ext.web.client.impl.HttpContext.next(HttpContext.java:365)
	at io.vertx.ext.web.client.impl.HttpContext.fire(HttpContext.java:332)
	at io.vertx.ext.web.client.impl.HttpContext.createRequest(HttpContext.java:223)
	at io.vertx.ext.web.client.impl.HttpContext.handlePrepareRequest(HttpContext.java:460)
	at io.vertx.ext.web.client.impl.HttpContext.execute(HttpContext.java:372)
	at io.vertx.ext.web.client.impl.HttpContext.next(HttpContext.java:365)
	at io.vertx.ext.web.client.impl.HttpContext.fire(HttpContext.java:332)
	at io.vertx.ext.web.client.impl.HttpContext.prepareRequest(HttpContext.java:211)
	at io.vertx.ext.web.client.impl.HttpRequestImpl.send(HttpRequestImpl.java:326)
	at io.vertx.ext.web.client.impl.HttpRequestImpl.sendBuffer(HttpRequestImpl.java:291)
	at io.apicurio.rest.client.VertxHttpClient.sendRequestWithPayload(VertxHttpClient.java:157)
	at io.apicurio.rest.client.VertxHttpClient.executePost(VertxHttpClient.java:121)
	at io.apicurio.rest.client.VertxHttpClient.sendRequest(VertxHttpClient.java:100)
	at io.apicurio.registry.rest.client.impl.RegistryClientImpl.createArtifact(RegistryClientImpl.java:236)
	at io.apicurio.registry.rest.client.RegistryClient.createArtifact(RegistryClient.java:138)
	at io.apicurio.registry.serde.DefaultSchemaResolver.lambda$handleAutoCreateArtifact$3(DefaultSchemaResolver.java:214)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1705)
	at io.apicurio.registry.serde.DefaultSchemaResolver.handleAutoCreateArtifact(DefaultSchemaResolver.java:213)
	at io.apicurio.registry.serde.DefaultSchemaResolver.resolveSchema(DefaultSchemaResolver.java:89)
	at io.apicurio.registry.serde.AbstractKafkaSerializer.serialize(AbstractKafkaSerializer.java:92)
	at org.apache.kafka.clients.producer.KafkaProducer.doSend(KafkaProducer.java:925)
	at org.apache.kafka.clients.producer.KafkaProducer.send(KafkaProducer.java:885)
	at io.smallrye.reactive.messaging.kafka.impl.ReactiveKafkaProducer.lambda$send$3(ReactiveKafkaProducer.java:115)
	at io.smallrye.context.impl.wrappers.SlowContextualConsumer.accept(SlowContextualConsumer.java:21)
	at io.smallrye.mutiny.operators.uni.builders.UniCreateWithEmitter.subscribe(UniCreateWithEmitter.java:22)
	... 5 more
```

Reproducer:

Update and compile your upstream Quarkus bits and then

`git clone https://github.com/quarkus-qe/quarkus-test-framework.git`
`mvn clean install -DskipTests` (in order to generate some stubs that you need to have in your local environment)
`mvn clean verify -Dopenshift -Pexamples -Dquarkus.platform.version=999-SNAPSHOT -pl examples/kafka-registry`

We don't have any issue in openshift with scenarios without AVRO/apicurio

**Comment by cescoffier** [comment]
@carlesarnal @EricWittmann  Isn't that an issue fixed in Apicurio?

**Comment by carlesarnal** [comment]
Issue fixed in apicurio, yes. I've opened a PR to upgrade the apicurio versions with the fixed ones https://github.com/quarkusio/quarkus/pull/21563.

**Comment by cescoffier** [comment]
Closed by #21563.
